include/rocksdb/filter_policy.h

```diff
@@ -41,6 +42,64 @@ class FilterBitsBuilder {
   // The return value of this function would be the filter bits,
   // The ownership of actual data is set to buf
   virtual Slice Finish(std::unique_ptr<const char[]>* buf) = 0;
+
+  // Calculate num of entries fit into a space.
+  virtual int CalculateNumEntry(const uint32_t space) {
+    throw new std::runtime_error("CalculateNumEntry not Implemented");
+  }
+};
+
+class FullFilterBitsBuilder : public FilterBitsBuilder {
```

Comment:
That is a big decision. So far the class was totally hidden. I moved here so I could at least test it. If we moved it to internal header then we would have an obligation to maintain it forever, for which there does not seem to be a strong motivation.